src/equations/ideal_glm_mhd_2d.jl

```diff
@@ -752,30 +849,30 @@ end
 end
 
 
-"""
-    noncons_interface_flux(u_left, u_right, orientation, mode, equations::IdealGlmMhdEquations2D)
-
-Strong form of non-conservative flux on a surface (Powell and GLM terms)
-```math
-phi^L 1/2 (B^L + B^R)_{normal} - phi^L B^L+{normal} = phi^L 1/2 (B^R - B^L)_{normal}
-```
-!!! note
-    The non-conservative interface flux depends on the discretization. Following "modes" are available:
-    * `:weak`: 'weak' formulation of split DG already includes the contribution
-      ``-1/2 (phi^L B^L_{normal})`` so this mode only adds ``1/2 (phi^L B^R_{normal})``,
-      analogously for the Galilean nonconservative term
-    * `:whole`: This mode adds the whole non-conservative term: phi^L 1/2 (B^R-B^L)
-    * `:inner`: This mode adds the split-form DG volume integral contribution: This is equivalent to
-      ``(2)-(1) - 1/2 (phi^L B^L)``
-!!! warning
-    This is non-unique along an interface! The normal direction is super important.
-
-For details see Section 4 of the paper
-- Bohm et al. (2018)
-  An entropy stable nodal discontinuous Galerkin method for the resistive MHD equations.
-  Part I: Theory and numerical verification
-  [DOI: 10.1016/j.jcp.2018.06.027](https://doi.org/10.1016/j.jcp.2018.06.027)
-"""
+
+# TODO: nonconservative terms, remove
+#     noncons_interface_flux(u_left, u_right, orientation, mode, equations::IdealGlmMhdEquations2D)
+
+# Strong form of non-conservative flux on a surface (Powell and GLM terms)
+# ```math
+# phi^L 1/2 (B^L + B^R)_{normal} - phi^L B^L+{normal} = phi^L 1/2 (B^R - B^L)_{normal}
+# ```
+# !!! note
+#     The non-conservative interface flux depends on the discretization. Following "modes" are available:
+#     * `:weak`: 'weak' formulation of split DG already includes the contribution
+#       ``-1/2 (phi^L B^L_{normal})`` so this mode only adds ``1/2 (phi^L B^R_{normal})``,
+#       analogously for the Galilean nonconservative term
+#     * `:whole`: This mode adds the whole non-conservative term: phi^L 1/2 (B^R-B^L)
+#     * `:inner`: This mode adds the split-form DG volume integral contribution: This is equivalent to
+#       ``(2)-(1) - 1/2 (phi^L B^L)``
+# !!! warning
+#     This is non-unique along an interface! The normal direction is super important.
+
+# For details see Section 4 of the paper
+# - Bohm et al. (2018)
+#   An entropy stable nodal discontinuous Galerkin method for the resistive MHD equations.
+#   Part I: Theory and numerical verification
+#   [DOI: 10.1016/j.jcp.2018.06.027](https://doi.org/10.1016/j.jcp.2018.06.027)
 @inline function noncons_interface_flux(u_left, u_right, orientation, mode,
```

Comment:
This version of the nonconservative flux that uses the `mode` is needed by the shock capturing. I believe that @amrueda has since streamlined this in FLUXO and removed the need for some of these modes in the FV fluxes. It would be good for him to weigh-in and say exactly what functionality we should keep around for the nonconservative terms + shock capturing on curved meshes in #736

Reply:
I checked the previous implementation of shock-capturing for MHD in Trixi.jl. My new version of handling nonconservative terms this PR is equivalent (in exact arithmetic) to the old one (on the `TreeMesh`). The old tests with our common strict tolerances pass without any modifications, so I'm pretty confident that everything is fine.
I haven't looked at your handling on shock-capturing with nonconservative terms on curvilinear meshes yet, so I would be happy to get more input from you all. Let's discuss this in the associated issue #736.

Reply:
Thanks for checking! I just wanted to be sure that we didn't lose some functionality in the updated nonconservative terms that would need to be put back in later

Reply:
Right now, we don't lose any feature we have in Trixi.jl so far.

Reply:
Yes, I implemented the non-conservative terms in Trixi.jl as I initially did in FLUXO. However, Flo and I later noticed that the modes :inner and :whole can be removed, as a cheaper and algebraically equivalent version of shock capturing can be done with only the :weak mode. I will be happy to update the Trixi,jl routines in #736 .

Reply:
That is great! So we can also simplify the implementation herein and actually remove the need for the `mode` in theses routines. I started a TODO list in #736 with this point.

Reply:
Well, I have already implemented a version that doesn't need these different modes at all in this PR - without looking at any FLUXO code. What remains to be done is the curvilinear variant.
Since the code using different modes is effectively only kept to avoid a breaking change right now, there is no need to change it. It would of course be nice to get another review of my new design of nonconservative terms in Trixi.jl - if you have time and want to do that